proposals/0013-selective-qualified-enum-access.md

```diff
@@ -0,0 +1,41 @@
+# ::Selective Qualified Enum Access::
+* Proposal: [HXP-0013](0013-selective-qualified-enum-access.md)
+* Author: [Haxe Developer](https://github.com/0b1kn00b)
+
+## Introduction
+
+Allow control over the publishing of Enum constructors in the global scope.
+
+## Motivation
+
+Namespacing is important because it allows natural and comprehensible names within any given scope. 
+
+Enum constructors break with this convention for the sake of brevity, but can cause problems when integrating several grammars and publishing artifacts. When publishing a programming library that requires the use of enums it's necessary to work around this flaw. 
+
+Given, that for simplicity sake, it's polite to require as few imports as possible; naming conventions must be applied so as not to cause aliasing, which nullifies the advantages of namespaces.
+
+Without those naming conventions, it's necessary to push enumerations into separate packages and then explain somewhere where the missing piece is.
+
+#### tldr:
+
+It should be possible to suggest that enum constructors be used qualified so as to respect the global scope.
+
+## Detailed design
+
+Using a metadata tag `@:qualified_enum_access` should cause unqualified references to it's constructors to be inaccessable. `Enum.Constructor` remains valid. 
+
+Possible override in the same manner of `@:access`: `@:unqualified_enum_access`
+
+## Impact on existing code
+  None I can think of.
+
+## Drawbacks
+  It would make some pattern matching code more verbose if the override isn't implemented, although `import LongEnumName in L` isn't too bad.
```

Comment:
This could be solved by allowing enum constructors to be explicitly imported into the local namespace using a wildcard:<sup>1</sup>

```haxe
import LongEnumName.*;
```

This way unqualified references would still be possible even for `@:qualified_enum_access` enums, and the constructors could even be exposed to entire particular packages/folders using `import.hx` files.

<sup>1</sup> Maybe also individually; even though that's less useful, it would be more consistent with importing other types.

---

P.S. It's basically how Rust handles enums: ~~constructors~~ variants are namespaced, but can be imported with wildcards; and commonly used variants are imported in preludes, of which the most notable is _the_ automatic Rust prelude that imports, among other things, the variants for `Option<T>` (`Some`, `None`) and `Result<T, E>` (`Ok`, `Err`).

Reply:
I'm easy as to how it looks myself, so long as a measure of control can be had beyond shuffling imports and so-forth. like it.